netty/src/main/java/io/grpc/netty/GrpcHttp2HeadersUtils.java

```diff
@@ -418,8 +421,12 @@ public CharSequence scheme() {
     public List<CharSequence> getAll(CharSequence csName) {
       AsciiString name = requireAsciiString(csName);
       if (isPseudoHeader(name)) {
-        // This code should never be reached.
-        throw new IllegalArgumentException("Use direct accessor methods for pseudo headers.");
+        AsciiString value = getPseudoHeader(name);
+        if (value == null) {
+          return Collections.emptyList();
+        } else {
+          return Collections.singletonList(value);
+        }
       }
       if (equals(TE_HEADER, name)) {
         return Collections.singletonList((CharSequence) te);
```

Comment:
I think this should also return empty list in not present case?

Reply:
If I had noticed when I found the bug with the pseudo headers, I might have fixed it. But I didn't, and it seems fine to be separate.